File: multiple-languages/dotnet/AlibabaCloud.SDK.ROS.CDK.Arms/AlibabaCloud/SDK/ROS/CDK/Arms/IEnvCustomJob.cs
using Amazon.JSII.Runtime.Deputy;

#pragma warning disable CS0672,CS0809,CS1591

namespace AlibabaCloud.SDK.ROS.CDK.Arms
{
    /// <summary>Represents a `EnvCustomJob`.</summary>
    [JsiiInterface(nativeType: typeof(IEnvCustomJob), fullyQualifiedName: "@alicloud/ros-cdk-arms.IEnvCustomJob")]
    public interface IEnvCustomJob : AlibabaCloud.SDK.ROS.CDK.Core.IResource
    {
        /// <summary>Attribute ConfigYaml: The YAML configuration string of the custom job.</summary>
        [JsiiProperty(name: "attrConfigYaml", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrConfigYaml
        {
            get;
        }

        /// <summary>Attribute EnvCustomJobName: The name of the custom job.</summary>
        [JsiiProperty(name: "attrEnvCustomJobName", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrEnvCustomJobName
        {
            get;
        }

        /// <summary>Attribute EnvironmentId: The ID of the environment instance.</summary>
        [JsiiProperty(name: "attrEnvironmentId", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
        object AttrEnvironmentId
        {
            get;
        }

        [JsiiProperty(name: "props", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-arms.EnvCustomJobProps\"}")]
        AlibabaCloud.SDK.ROS.CDK.Arms.IEnvCustomJobProps Props
        {
            get;
        }

        /// <summary>Represents a `EnvCustomJob`.</summary>
        [JsiiTypeProxy(nativeType: typeof(IEnvCustomJob), fullyQualifiedName: "@alicloud/ros-cdk-arms.IEnvCustomJob")]
        internal sealed class _Proxy : DeputyBase, AlibabaCloud.SDK.ROS.CDK.Arms.IEnvCustomJob
        {
            private _Proxy(ByRefValue reference): base(reference)
            {
            }

            /// <summary>Attribute ConfigYaml: The YAML configuration string of the custom job.</summary>
            [JsiiProperty(name: "attrConfigYaml", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrConfigYaml
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute EnvCustomJobName: The name of the custom job.</summary>
            [JsiiProperty(name: "attrEnvCustomJobName", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrEnvCustomJobName
            {
                get => GetInstanceProperty<object>()!;
            }

            /// <summary>Attribute EnvironmentId: The ID of the environment instance.</summary>
            [JsiiProperty(name: "attrEnvironmentId", typeJson: "{\"union\":{\"types\":[{\"primitive\":\"string\"},{\"fqn\":\"@alicloud/ros-cdk-core.IResolvable\"}]}}")]
            public object AttrEnvironmentId
            {
                get => GetInstanceProperty<object>()!;
            }

            [JsiiProperty(name: "props", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-arms.EnvCustomJobProps\"}")]
            public AlibabaCloud.SDK.ROS.CDK.Arms.IEnvCustomJobProps Props
            {
                get => GetInstanceProperty<AlibabaCloud.SDK.ROS.CDK.Arms.IEnvCustomJobProps>()!;
            }

            /// <summary>The environment this resource belongs to.</summary>
            /// <remarks>
            /// For resources that are created and managed by the CDK
            /// (generally, those created by creating new class instances like Role, Bucket, etc.),
            /// this is always the same as the environment of the stack they belong to;
            /// however, for imported resources
            /// (those obtained from static methods like fromRoleArn, fromBucketName, etc.),
            /// that might be different than the stack they were imported into.
            /// </remarks>
            [JsiiProperty(name: "env", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-core.IResourceEnvironment\"}")]
            public AlibabaCloud.SDK.ROS.CDK.Core.IResourceEnvironment Env
            {
                get => GetInstanceProperty<AlibabaCloud.SDK.ROS.CDK.Core.IResourceEnvironment>()!;
            }

            /// <summary>The stack in which this resource is defined.</summary>
            [JsiiProperty(name: "stack", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-core.Stack\"}")]
            public AlibabaCloud.SDK.ROS.CDK.Core.Stack Stack
            {
                get => GetInstanceProperty<AlibabaCloud.SDK.ROS.CDK.Core.Stack>()!;
            }

            /// <summary>The construct tree node for this construct.</summary>
            [JsiiProperty(name: "node", typeJson: "{\"fqn\":\"@alicloud/ros-cdk-core.ConstructNode\"}")]
            public AlibabaCloud.SDK.ROS.CDK.Core.ConstructNode Node
            {
                get => GetInstanceProperty<AlibabaCloud.SDK.ROS.CDK.Core.ConstructNode>()!;
            }
        }
    }
}

File: multiple-languages/go/alicloudroscdkarms/IEnvCustomJob.go
package alicloudroscdkarms

import (
	_jsii_ "github.com/aws/jsii-runtime-go/runtime"

	"github.com/alibabacloud-go/ros-cdk/alicloudroscdkarms/internal"
	"github.com/alibabacloud-go/ros-cdk/alicloudroscdkcore"
)

// Represents a `EnvCustomJob`.
type IEnvCustomJob interface {
	alicloudroscdkcore.IResource
	// Attribute ConfigYaml: The YAML configuration string of the custom job.
	AttrConfigYaml() interface{}
	// Attribute EnvCustomJobName: The name of the custom job.
	AttrEnvCustomJobName() interface{}
	// Attribute EnvironmentId: The ID of the environment instance.
	AttrEnvironmentId() interface{}
	Props() *EnvCustomJobProps
}

// The jsii proxy for IEnvCustomJob
type jsiiProxy_IEnvCustomJob struct {
	internal.Type__alicloudroscdkcoreIResource
}

func (j *jsiiProxy_IEnvCustomJob) AttrConfigYaml() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrConfigYaml",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IEnvCustomJob) AttrEnvCustomJobName() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrEnvCustomJobName",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IEnvCustomJob) AttrEnvironmentId() interface{} {
	var returns interface{}
	_jsii_.Get(
		j,
		"attrEnvironmentId",
		&returns,
	)
	return returns
}

func (j *jsiiProxy_IEnvCustomJob) Props() *EnvCustomJobProps {
	var returns *EnvCustomJobProps
	_jsii_.Get(
		j,
		"props",
		&returns,
	)
	return returns
}

